src/main/java/io/vertx/core/impl/launcher/commands/StartCommand.java

```diff
@@ -0,0 +1,153 @@
+public class StartCommand extends DefaultCommand {
+
+  private String id;
+  private String launcher;
+
+  /**
+   * Sets the "application id" that would be to stop the application and be lsited in the {@code list} command.
+   *
+   * @param id the id
+   */
+  @Option(longName = "vertx.id", shortName = "id", required = false, acceptValue = true)
+  @Description("The id of the application, a random UUID by default")
+  public void setApplicationId(String id) {
+    this.id = id;
+  }
+
+  /**
+   * A hidden option to set the launcher class.
+   *
+   * @param clazz the class
+   */
+  @Option(longName = "launcher-class")
+  @Hidden
+  public void setLauncherClass(String clazz) {
+    this.launcher = clazz;
+  }
+
+  /**
+   * Starts the application in background.
+   */
+  @Override
+  public void run() {
+    out.println("Starting vert.x application...");
+    List<String> cmd = new ArrayList<>();
+    ProcessBuilder builder = new ProcessBuilder();
+    addJavaCommand(cmd);
+
+    // Add the classpath to env.
+    builder.environment().put("CLASSPATH", System.getProperty("java.class.path"));
+
+    if (launcher != null) {
+      ExecUtils.addArgument(cmd, launcher);
+    } else if (isLaunchedAsFatJar()) {
+      ExecUtils.addArgument(cmd, "-jar");
+      ExecUtils.addArgument(cmd, CommandLineUtils.getJar());
+    } else {
+      ExecUtils.addArgument(cmd, CommandLineUtils.getFirstSegmentOfCommand());
+    }
+
+    getArguments().stream().forEach(arg -> ExecUtils.addArgument(cmd, arg));
+
+    try {
+      builder.command(cmd);
+      builder.start();
+      out.println(id);
+    } catch (IOException e) {
+      out.println("Cannot create vert.x application process");
+      e.printStackTrace(out);
+    }
+
+  }
+
+  private void addJavaCommand(List<String> cmd) {
+    if (ExecUtils.isWindows()) {
+      ExecUtils.addArgument(cmd, "cmd.exe");
+      ExecUtils.addArgument(cmd, "/C");
+      ExecUtils.addArgument(cmd, "start");
+      ExecUtils.addArgument(cmd, "vertx.id - " + id);
+      ExecUtils.addArgument(cmd, "/B");
+    }
+    ExecUtils.addArgument(cmd, getJava().getAbsolutePath());
+  }
+
+  private File getJava() {
+    File java;
+    File home = new File(System.getProperty("java.home"));
+    if (ExecUtils.isWindows()) {
+      java = new File(home, "bin/java.exe");
+    } else {
+      java = new File(home, "bin/java");
+    }
+
+    if (!java.isFile()) {
+      throw new IllegalStateException("Cannot find java executable - " + java.getAbsolutePath()
+          + " does not exist");
+    }
+    return java;
+  }
+
+  private boolean isLaunchedAsFatJar() {
+    return CommandLineUtils.getJar() != null;
+  }
+
+  private List<String> getArguments() {
+    List<String> args = executionContext.commandLine().allArguments();
+    // Add system properties passed as parameter
+    if (systemProperties != null) {
+      args.addAll(
```

Comment:
seems simpler to do `systemProperties.stream().map(entry -> "-D" + entry).forEach(args::add)`